### ISSUE TITLE
recording metrics through FG

### ISSUE THREAD

**Issue by raynardj** [raynardj/forge#17]
### The function in the ```class forgedb``` will include
* [x] registering new metric
* [x] keep a record in the log
* [x] check if this is the best performance

### The function in the ftorch
* [x] collect Trainer result(a dictionary) automatically

**Comment by raynardj** [comment]
if Trainer(..... callbacks = [fg.metrics()]) was set, the metrics will be saved automatically
